File: public/tools/elements/profile/ProfileUser.ts
import {Element, Property} from "../../polymer/Annotations";
import {GtBox} from "../widgets/GtBox";
import {BnetThumb} from "../misc/BnetThumb";
import {PolymerElement} from "../../polymer/PolymerElement";

@Element({
	selector: "profile-user",
	template: "/assets/views/profile.html",
	dependencies: [GtBox, BnetThumb]
})
export class ProfileUser extends PolymerElement {
	@Property public user: number;
}

File: client/src/gt/component/profile/ProfileUser.scala
package gt.component.profile

import gt.component.GtHandler
import gt.component.widget.{BnetThumb, GtBox}
import gt.service.Roster
import util.jsannotation.js
import xuen.Component

object ProfileUser extends Component[ProfileUser](
	selector = "profile-user",
	templateUrl = "/assets/imports/views/profile.html",
	dependencies = Seq(GtBox, BnetThumb)
)

@js class ProfileUser extends GtHandler {
	val user = property[Int]
	val roster = service(Roster)

	val rank = user ~! roster.user ~ (_.group)
	val main = user ~! roster.main
}
